internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,163 @@ import (
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
 )
 
+
+
+// Runtime2 is the VM.
+// Review: the object that implements this outlives actor calls.
+type Runtime2 struct {
+	CurerntEpoch() Epoch
+	// Review: why is this taking an offset? verify size of offset
+	Randomness(epoch Epoch, offset uint64) Randomness
+	// Review: the spec still has a TODO on this
+	// Review: not sure why the original has a messagereceipt instead of an InvocOutput
+	Send(input InvocInput) InvocOutput
+	// Review: how does this work? What qualifies as an "error"?
+	// Review: how is this any different than the caller just ignoring a returned error?
+	// SendAllowingErrors(input msg.InvocInput) msg.MessageReceipt
+
+	// Review: we might want a global Storage() Storage here
```

Comment:
What would be the use case -- i.e. what data would be stored here (outside of the actor's own state in the state tree)?

Reply:
It is more of a question for the spec team.. not sure what might be needed by the actors. If the answer is nothing then we don't need it now, it easy to add later. 

From my viewpoint, I see the chain as something immutable present in the global store for actor to potentially read pieces of.

Btw this only makes sense if the local-store the actor gets is JUST for things the actor wrote himself. If there is not actor-local store, then the only thing we have IS a global store.

Reply:
See below -- I don't think the isolation property is critical.